src/Microsoft.VisualBasic.Core/tests/DateTypeTests.cs

```diff
@@ -0,0 +1,218 @@
+            yield return new object[] { (UIntEnum)1, default(DateTime) };
+
+            // int.
+            yield return new object[] { int.MinValue, default(DateTime) };
+            yield return new object[] { -1, default(DateTime) };
+            yield return new object[] { 0, default(DateTime) };
+            yield return new object[] { 1, default(DateTime) };
+            yield return new object[] { int.MaxValue, default(DateTime) };
+            yield return new object[] { (IntEnum)int.MinValue, default(DateTime) };
+            yield return new object[] { (IntEnum)(-1), default(DateTime) };
+            yield return new object[] { (IntEnum)0, default(DateTime) };
+            yield return new object[] { (IntEnum)1, default(DateTime) };
+            yield return new object[] { (IntEnum)int.MaxValue, default(DateTime) };
+
+            // ulong.
+            yield return new object[] { ulong.MinValue, default(DateTime) };
+            yield return new object[] { (ulong)1, default(DateTime) };
+            yield return new object[] { (ULongEnum)ulong.MinValue, default(DateTime) };
+            yield return new object[] { (ULongEnum)1, default(DateTime) };
+
+            // long.
+            yield return new object[] { (long)(-1), default(DateTime) };
+            yield return new object[] { (long)0, default(DateTime) };
+            yield return new object[] { (long)1, default(DateTime) };
+            yield return new object[] { (LongEnum)(-1), default(DateTime) };
+            yield return new object[] { (LongEnum)0, default(DateTime) };
+            yield return new object[] { (LongEnum)1, default(DateTime) };
+
+            // float.
+            yield return new object[] { (float)(-1), default(DateTime) };
+            yield return new object[] { (float)0, default(DateTime) };
+            yield return new object[] { (float)1, default(DateTime) };
+
+            // double.
+            yield return new object[] { (double)(-1), default(DateTime) };
+            yield return new object[] { (double)0, default(DateTime) };
+            yield return new object[] { (double)1, default(DateTime) };
+
+            // decimal.
+            yield return new object[] { (decimal)(-1), default(DateTime) };
+            yield return new object[] { (decimal)0, default(DateTime) };
+            yield return new object[] { (decimal)1, default(DateTime) };
+
+            // bool.
+            yield return new object[] { true, default(DateTime) };
+            yield return new object[] { false, default(DateTime) };
+        }
+
+        private static IEnumerable<object[]> FromObject_Invalid_TestData()
+        {
+            yield break;
+        }
+
+        private static IEnumerable<object[]> FromObject_Overflow_TestData()
+        {
+            yield break;
+        }
+
+        private static IEnumerable<object[]> FromString_TestData()
+        {
+            yield return new object[] { "12:00:00 AM", new DateTime(0) };
```

Comment:
Consider having one more test to verify that it doesn't always return `default`.